src/zope/schema/_field.py

```diff
@@ -214,7 +215,13 @@ def fromUnicode(self, uc):
         try:
             v = decimal.Decimal(uc)
         except decimal.InvalidOperation:
-            raise ValueError('invalid literal for Decimal(): %s' % uc)
+            exc = ValueError('invalid literal for Decimal(): %s' % uc)
+            exc.value = uc
+            exc.field = self
+            try:
+                raise exc
+            finally:
+                del exc
```

Comment:
This is a bit ugly.  I would prefer a custom ValidationError subclass (that also subclasses ValueError) to be used here.

Reply:
Good point, will do.

Reply:
``Float`` and ``Int`` needed the same treatment. I didn't make any of those exception classes public, but their names show up in the documentation (because it's doctests). I wonder if I should?